src/engine/MaxConstraint.cpp

```diff
@@ -284,6 +288,8 @@ void MaxConstraint::resetMaxIndex()
     double maxValue = FloatUtils::negativeInfinity();
     _maxIndexSet = false;
 
+    // Guy: it looks to me like the max index should be set even if
+    // only _f is assigned, if _f is in _elements. What do you think?
```

Comment:
Yes, I should have fix this as well and will do.

Reply:
Okay, so lets fix this before we merge the PR.

Reply:
I fixed this.